src/main/webapp/itempagescript.js

```diff
@@ -12,15 +12,75 @@ async function loadItemPage() {
       fetch(`/itempagedata?itemId=${itemId}`)
           .then((response) => response.json())
           .then((itemPageData) => {
-            createSelectedItemCard(itemPageData.item);
-            getItemPageComments(itemPageData.comments);
+            fetch('/login')
+                .then((response) => response.json())
+                .then((loginResponse) => {
+                  if (loginResponse.isUserLoggedIn) {
+                    createSelectedItemCard(itemPageData.item);
+                    getItemPageComments(itemPageData.comments);
```

Comment:
```
createSelectedItemCard(itemPageData.item);
getItemPageComments(itemPageData.comments);
```
 gets called in both the `if` body and the `else` body. So you can just move it out before the if/else block.